### PR TITLE
Use bazel 3.4.1 in test-infra

### PR DESCRIPTION
**Pull request by fejta**
* Followed migration instructions at https://github.com/bazelbuild/rules_nodejs/wiki#migrating-to-30
  - changed how `jasmine_node_test`, `rollup_bundle`, `ts_library` etc are imported
  - a few changes to how these are called
  - added `common --experimental_allow_incremental_repository_updates` to bazelrc per instructions
* Updated github.com/bazelbuild/buildtools to v0.0.0-20200922170545-10384511ce98 (matching repo-infra)
  - Required updating buildifier plugin as interface changed
* Removed --incompatible_disable_deprecated_attr_params=false from bazelrc that no longer exists at 3.4.1

### CODE REVIEW

**Comment by fejta** [comment]
> ERROR: The project you're trying to build requires Bazel 3.4.1 (specified in /home/prow/go/src/github.com/kubernetes/test-infra/.bazelversion), but it wasn't found in /usr/local/lib/bazel/bin.

https://prow.k8s.io/prowjob?prowjob=8381c680-61b3-11eb-8203-be7ec3c1d3e8 shows 
> gcr.io/k8s-testimages/launcher.gcr.io/google/bazel:v20201028-8000225-test-infra

Will need to wait for https://github.com/kubernetes/test-infra/pull/20649

**Comment by fejta** [comment]
Once that PR merges we'll just be able to `/retest` here and tests should pass (so feel free to review now)

**Review by cjwagner — APPROVED** [review]
/hold

**Comment by fejta** [comment]
Oh right, this error isn't repoing locally...

**Comment by fejta** [comment]
This is kind of a slog, think I need to follow this more closely: https://github.com/bazelbuild/rules_nodejs/wiki#migrating-off-internalrollup

currently getting:

```
==================== Test output for //prow/cmd/deck/static:prow_test:
internal/modules/cjs/loader.js:797
    throw err;
    ^

Error: Cannot find module 'io_k8s_test_infra/prow/cmd/deck/static/prow/histogram'
Require stack:
- /private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/io_k8s_test_infra/prow/cmd/deck/static/prow/histogram_test.js
- /private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/npm/node_modules/jasmine/lib/jasmine.js
- /private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/npm/node_modules/@bazel/jasmine/index.js
- /private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/npm/node_modules/@bazel/jasmine/jasmine_runner.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:794:15)
    at Function.Module._load (internal/modules/cjs/loader.js:687:27)
    at Module.require (internal/modules/cjs/loader.js:849:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at /private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/io_k8s_test_infra/prow/cmd/deck/static/prow/histogram_test.js:13:25
    at /private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/io_k8s_test_infra/prow/cmd/deck/static/prow/histogram_test.js:3:17
    at Object.<anonymous> (/private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/io_k8s_test_infra/prow/cmd/deck/static/prow/histogram_test.js:9:3)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Module.require (internal/modules/cjs/loader.js:849:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at /private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/npm/node_modules/jasmine/lib/jasmine.js:89:5
    at Array.forEach (<anonymous>)
    at Jasmine.loadSpecs (/private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/npm/node_modules/jasmine/lib/jasmine.js:88:18)
    at Jasmine.execute (/private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/npm/node_modules/jasmine/lib/jasmine.js:259:8)
    at main (/private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/npm/node_modules/@bazel/jasmine/jasmine_runner.js:157:11)
    at Object.<anonymous> (/private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/npm/node_modules/@bazel/jasmine/jasmine_runner.js:179:22)
    at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
    at internal/main/run_main_module.js:17:11 {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/io_k8s_test_infra/prow/cmd/deck/static/prow/histogram_test.js',
    '/private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/npm/node_modules/jasmine/lib/jasmine.js',
    '/private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/npm/node_modules/@bazel/jasmine/index.js',
    '/private/var/tmp/_bazel_fejta/436df7e80b7a8ba049d812837a328382/sandbox/darwin-sandbox/6/execroot/io_k8s_test_infra/bazel-out/darwin-fastbuild/bin/prow/cmd/deck/static/prow_test.sh.runfiles/npm/node_modules/@bazel/jasmine/jasmine_runner.js'
  ]
}
```

**Comment by fejta** [comment]
WORKS ON MY MACHINE (TM)

**Comment by fejta** [comment]
WHAT! verify-codegen?? I disagree with this failure! 😇

**Comment by fejta** [comment]
HA! TAKE THAT `pull-test-infra-bazel` https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/20651/pull-test-infra-bazel/1357236727990718464

File: prow/plugins/buildifier/buildifier.go
Comment (chaodaiG) [inline prow/plugins/buildifier/buildifier.go:129]: Not quite sure why this is also part of the PR, is it intended?

File: repos.bzl
Comment (chaodaiG) [inline repos.bzl:18]: I hope this file was generated

File: def/ts.bzl
Comment (chaodaiG) [inline def/ts.bzl:22]: Is there anything need to be documented here explaining why combining `rollup_bundle` with `terser_minified`?

**Review by chaodaiG — COMMENTED** [review]
Overall LGTM, left a nit and a question

File: prow/plugins/buildifier/buildifier.go
Comment (fejta) [reply prow/plugins/buildifier/buildifier.go:129]: Yes, `build.Rewrite()` no longer exists at the updated version of github.com/bazelbuild/buildtools/build that repo-infra uses.

File: def/ts.bzl
Comment (fejta) [reply def/ts.bzl:22]: Just following https://github.com/bazelbuild/rules_nodejs/wiki#migrating-off-internalrollup instructions

File: repos.bzl
Comment (fejta) [reply repos.bzl:18]: Yes, this is what `update-deps` generates (via gazelle)

**Pull request merged**
